src/operator/softmax_output-inl.h

```diff
@@ -0,0 +1,476 @@
+/*
```

Comment:
These files were removed in https://github.com/apache/incubator-mxnet/pull/18531
What's the intention of adding them back? If you like to reconsider the plan for supporting the Module API in MXNet 2, why just add back SoftmaxOutput and not the other loss function? Should this be discussed in the https://github.com/apache/incubator-mxnet/issues/16167 RFC (ie. providing an update that you want to support the Module API?)